src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputer.java

```diff
@@ -117,9 +123,18 @@ public HttpResponse doDoDelete() throws IOException {
         ComputeEngineInstance node = getNode();
         if (node != null) {
             try {
+                ComputeEngineCloud cloud = getCloud();
+
+                // Checks for failed jobs for this computer's node
+                if (cloud != null && !this.getBuilds().failureOnly().isEmpty()) {
```

Comment:
I think that should be setting whenever we want to save snapshots or not..

Reply:
This is in the next PR.